### DISCUSSION TITLE
Looking for ways to set a custom MTU for a network

### DISCUSSION THREAD

**Discussion by mfld-pub** [dec0dOS/zero-ui#98] [Q&A]
Have some sites in the mix where the WAN MTU is something exotic.

When I join these sites to ZT networks issues are observed that indicate MTU problems.

i.e. pings work but transfers of large data fail. SSH login works but freezes during banner display. 

I found my API key in db.json

```
cat /var/lib/docker/volumes/root_zero-ui_data/_data/db.json | grep token
      "token": "mysuperawesometoken"
```

The token seems to work - I can interact with the API:

`curl -s -H "Authorization: Bearer mysuperawesometoken"  -H "Content-Type: application/json"      https://localhost/api/network      |  jq -r '.[]| "\(.config.name)***\(.id)"' | column -t -s "***"`

works and displays my networks. 

But when I try to force a network ID's MTU to something custom it will seem to accept it but it doesn't take effect.

I try doing it like this:

`curl -X POST "https://localhost/api/network/somenetworkid" -H "Authorization: Bearer mysuperawesometoken" -d '{"config": {"mtu": 1432}}'`

When I check, all MTU are showing as 2800.

```
curl -s -H "Authorization: Bearer mysuperawesometoken"  -H "Content-Type: application/json"      https://localhost/api/network      | jq | grep mtu
      "mtu": 2800,
      "mtu": 2800,
      "mtu": 2800,
      "mtu": 2800,
      "mtu": 2800,
      "mtu": 2800,


```

EDIT: Just noticed that zu-main container restarts when I try to post to the API endpoint to modify an MTU.

```
/app/backend/services/network.js:78
  if (data.hasOwnProperty("description")) {
           ^

TypeError: data.hasOwnProperty is not a function
    at Object.updateNetworkAdditionalData (/app/backend/services/network.js:78:12)
    at /app/backend/routes/network.js:59:11
    at Layer.handle [as handle_request] (/app/backend/node_modules/express/lib/router/layer.js:95:5)
    at next (/app/backend/node_modules/express/lib/router/route.js:144:13)
    at isAuthorized (/app/backend/services/auth.js:29:9)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

Will likely turn this into a bug or feature request. _Should_ I be able to set a custom MTU with zero-ui or is it not currently a feature ?

**Comment by dec0dOS** [comment]
Hello, @mfld-pub! Thanks for your report. As the MTU change does not present in the UI (also as in the ZeroTier Central), it wasn't properly tested. It could also be the bug as in the ZeroUI backend, or in the ZeroTier controller itself on specific version.

The presented error is really strange, as it should not fail there.
